doc/release.md

```diff
@@ -27,21 +23,41 @@ git commit -am "Upgrade the version to x.y.z+1-dev0 following x.y.z release"
 
 Check the result and then upgrade the main branch
 
-### Milestone handling
-
-We move issue that were not done in the next milestone and block release only if it's an
-issue labelled as blocker.
+9. Delete the `X.Y-1` branch.
 
-## Post release
+## Release a patch version
 
-### Merge tags in main for pre-commit
+We release patch version when a crash or a bug is fixed on the main branch, and it needs
+to be backported.
 
-If the tag you just made is not part of the main branch, merge the tag `vX.Y.Z` in the
-main branch by doing a history only merge. It's done in order to signal that this is an
-official release tag, and for `pre-commit autoupdate` to works.
+1. During the merge request on main, make sure that the changelog is for the patch
+   version `X.Y-1.Z'`.
+2. (Create a `X.Y-1` branch from the `X.Y-1.0` tag if it does not already exist.)
+3. After the PR is merged cherry-pick the commits on the `X.Y-1` branch
+4. Check the result of `git diff vX.Y-1.Z-1 ChangeLog`.
+5. Install the release dependencies `pip3 install pre-commit tbump`
+6. Bump the version and release by using `tbump X.Y-1.Z --no-push`.
+7. Check the result visually and then by triggering the "release tests" workflow in
+   GitHub Actions first.
+8. Push the tag.
+9. Release the version on GitHub with the same name as the tag and copy and paste the
+   appropriate changelog in the description. This trigger the pypi release.
+10. Move back to a dev version with `tbump`:
 
 ```bash
-git checkout main
-git merge --no-edit --strategy=ours vX.Y.Z
-git push
+tbump X.Y-1.Z+1-dev0 --no-tag --no-push # You can interrupt during copyrite
+git commit -am "Upgrade the version to x.y-1.z+1-dev0 following x.y-1.z release"
 ```
+
+Check the result and then upgrade the `X.Y-1` branch
+
+13. Merge the `X.Y-1` branch on the main branch. The main branch should have the
+    changelog for `X.Y-1.Z+1`. This merge is required so `pre-commit autoupdate` works
+    for pylint.
+14. Fix version conflicts properly, or bump the version to `X.Y.0-devZ` before pushing
+    on the main branch
```

Comment:
```suggestion
7. Create a new branch based on the `main` branch.
8. Merge the newly created tag into the branch: `git merge --no-edit --strategy=ours vX.Y-1.Z`. (For example: `git merge --no-edit --strategy=ours v2.4.1`.
9. Merge the changes to changelog, contributors and documentation from the maintenance branch into the branch. **Note: make sure that all changelog entries included in the patch release are removed from the changelog for the in-development version on `main`.**
10. Make a PR from the new branch on `main`.

The last steps are necessary so that `main` has access to the correct tags as well as a consolidated changelog history.
```

Reply:
Ha I understand why you don't want to merge the fix branch. You want to remove the patch release note from the main branch ? But we have only one documentation online, we want to have all the release notes on it for all versions in the right version. If we backport fix in the 2.10.x branch the first time something is released will be 2.10.1 or something, but it will also be included in 2.11.0. If we remove those release note it won't be in the release notes at all (the main branch being the source of truth for the online doc). Thus we can simply merge.

Reply:
I think it will be on `main`.

What I propose is:
1) Initial PR targets `2.5.0` changelog 
_When you open the PR you cannot determine which patch release it will go in, as we are now probably releasing those more often. Therefore, you should just target the in-development version and let the patch version releaser deal with moving the changelog entry_
2) After the cherry-pick a commit is added that moves the changelog entry from `2.5.0` to `2.4.1`
3) After `2.4.1` is released a post-release PR is made to `main` which updates changelog, contributors and docs. Thereby also moving the changelog entry from `2.5.0` to `2.4.1` on `main`.

At the end the changelog entry will thus be in the version that first released it. Doing it this way makes it so that updating the changelog position becomes the responsibility of the patch version releaser (who only has to do it once) instead of the PR opener (who will have to update his PR every time a new patch release is released).
For `astroid` the difference might be quite low as we don't release patches very often, but for `pylint` not having to update all PRs after every patch release is quite useful I think.

Reply:
I see !

I think patch releases would still be "blocked" until we have ~~all~~ most the flying bug/crash PR done so it should not happen very often that you need to rebase and change the location in the changelog. But if it does then we don't need to do it, I'll fix it at release time like I currently do (this is what the ``git diff changelog v2.4.0`` is about, I check that all modification to the changelog are in the proper section). I'd rather have some release note to fix than all release note to fix. Also the decision to backport should be taken in the MR.  When we merge the fix branch in main, it will also fix the ``main`` branch. 

What we can do to make this process easier is to create a "To be backported" section that we then copy pasted later when releasing (Marc talked about that being done in black)

Reply:
>  Also the decision to backport should be taken in the MR. When we merge the fix branch in main, it will also fix the `main` branch.

This is also fine with me, but note that this is a different process than the one we have been discussing so far.

As I interpreted it, we either:
Commit to `main`, cherry-pick to `maintenance`, push changelog/doc and the tag to `main` from `maintenance`
Or:
Commit to `maintenance`, merge `maintenance` in `main`. 

I thought the whole idea was not to merge `maintenance` into `main` and only push the tags into `main`. Otherwise any fix-PR can just target the `maintenance` branch, as it will eventually end up in `main`. There is no reason to cherry-pick in that case.

But okay, we have been spending way too much time discussing this and you're likely the one who has to fix any issues that come from this process. I just wanted to highlight (and potentially solve) some future issues that might arise from this not being crystal-clear. As long as you have a clear idea of how you want to approach things I guess it should be fine.

Reply:
I did not apply the suggestion directly in github because there was a lot to change but I applied it manually.